<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="audio_topic" default="/rero_ros/audio_stream"/>
  <arg name="voice_detected_topic" default="/vad_voice_detected"/>
  <arg name="model_path" default="/home/turtle/ws/rero_ros/models/vad/silero_vad.onnx"/>
<arg name="silence_to_voice_threshold" default="0.4"/>
<arg name="voice_to_silence_threshold" default="0.3"/>
<arg name="min_silence_duration_ms" default="250"/>
  <arg name="debug_every_n_frames" default="5"/>
  <arg name="num_channels" default="4"/>
  <arg name="enable_audio_stats" default="true"/>
  <arg name="volume_analysis_frames" default="100"/>

  <group>
    <push-ros-namespace namespace="rero_ros"/>
    <node pkg="rero_ros" name="vad_node" exec="vad_node" output="screen">
      <param name="audio_topic" value="$(var audio_topic)"/>
      <param name="voice_detected_topic" value="$(var voice_detected_topic)"/>
      <param name="model_path" value="$(var model_path)"/>
      <param name="silence_to_voice_threshold" value="$(var silence_to_voice_threshold)"/>
      <param name="voice_to_silence_threshold" value="$(var voice_to_silence_threshold)"/>
      <param name="min_silence_duration_ms" value="$(var min_silence_duration_ms)"/>
      <param name="debug_every_n_frames" value="$(var debug_every_n_frames)"/>
      <param name="num_channels" value="$(var num_channels)"/>
      <param name="enable_audio_stats" value="$(var enable_audio_stats)"/>
      <param name="volume_analysis_frames" value="$(var volume_analysis_frames)"/>
    </node>
  </group>
</launch>
